app/api/admin_statistics_api/users.py

```diff
@@ -47,34 +46,26 @@ def verified_count(self, obj):
     def unverified_count(self, obj):
         return get_count(User.query.filter_by(is_verified=False, is_super_admin=False, is_admin=False))
 
-    def get_all_user_roles(self, role_name):
+    def get_all_user_role_count(self, role_name):
         role = Role.query.filter_by(name=role_name).first()
-        uers = UsersEventsRoles.query.join(UsersEventsRoles.event).join(UsersEventsRoles.role).filter(
-            Event.deleted_at.is_(None), UsersEventsRoles.role == role)
-        return uers
-
-    def get_distinct_roles_count(self, query_result):
-        """
-        Returns the length of set of users for the given query
-        """
-        distinct_entry = set([i.user for i in query_result])
-        return len(distinct_entry)
+        newquery = User.query.join(UsersEventsRoles.user).join(UsersEventsRoles.role).filter(
+            UsersEventsRoles.role==role).distinct()
+        counter = 0
+        for element in newquery:
+            counter += 1
```

Comment:
Still the same problem as before but much worse

Reply:
@iamareebjamal could you please elaborate on the similarity and the approach to be adopted for tackling this issue?

I am using `distinct` method for the query already, how do you want me to optimise it. I would really appreciate your guidance here

Reply:
See the get_count method. You should not use any loop or set of any other pythonic construct to get the count from the database

Reply:
@iamareebjamal oh, so the counting is the problem. Understood, I shall look into it

Reply:
@iamareebjamal Better now?